up.go

```diff
@@ -45,6 +68,24 @@ func UpTo(db *sql.DB, dir string, version int64, opts ...OptionsFunc) error {
 		return err
 	}
 
+	switch option.lockMode {
+	case LockModeAdvisorySession:
+		conn, err := db.Conn(ctx)
```

Comment:
This connection should be re-used in the event `*sql.DB` was configured with a maximum of 1 open connection. Otherwise, this will block indefinitely.